### ISSUE TITLE
Call to a member function read() on string in ObjectSerializer.php:285 

### ISSUE THREAD

**Issue by BelloAM** [docusign/docusign-php-client#102]
I get this error whenever i run the EnvelopeApi\getDocument method. It seems that the $data being passed to the deserialize function is a string.

I managed to get it to work by changing line  285-287 to 
```
if(is_string($data)){
    fwrite($file, $data);
}
else{
    while ($chunk = $data->read(200)) {
        fwrite($file, $chunk);
    } 
}
```
Unfortunately; this change is always removed when we deploy to our servers

**Comment by LarryKlugerDS** [comment]
Thank you @BelloAM  for this problem report.
I have filed bug report DCM-4285

**Comment by LarryKlugerDS** [comment]
Note that the EnvelopeApi\getDocument method should be updated to not write to a local file. See issue #56

**Comment by fdiedler** [comment]
Hi,
I have the same problem since one week. When this bug will be fixed ? 
Thanks

**Comment by LarryKlugerDS** [comment]
Unfortunately we don't yet have a fix scheduled from the engineering group. I have asked them again.

**Comment by fdiedler** [comment]
I am very embarrassed with this bug, we cannot retrieve contract file anymore and it blocks the onboarding of new clients in my company... Please correct it as soom as possible and thanks for asking them again about this bug. I think I am not the only people affected :)

Otherwise, It is possible to downgrade to a prior version withour the bug ? And if so, which version ? 

Thanks, and looking forward the fix

**Comment by LarryKlugerDS** [comment]
I am sorry that  this is causing a problem for you. 
The prior releases of the SDK are available from [GitHub](https://github.com/docusign/docusign-php-client/releases). They are also available from the [PHP Packagist site](https://packagist.org/packages/docusign/esign-client).

**Comment by fdiedler** [comment]
Aie, I think I cannot downgrade because I use a wrapper : 
tucker-eric/docusign-rest-client

So, I just removed the code that calls getDocument() method and I will wait for a fix and hope it will be soon :)

Thanks,

**Comment by wehostadm** [comment]
Hi,

Just to say that I am always waiting for a fix for this bug as soon as possible.
It will be a shame and a waste of time if I need to recode the function getDocument()

Maybe use the quick fix of @BelloAM above and then do a better fix if needed ?

Thanks a lot for this DocuSign bundle.

**Comment by LarryKlugerDS** [comment]
I've escalated the bug, but we don't yet have a schedule for the fix.

**Comment by cgsmith** [comment]
Hi @LarryKlugerDS 

Are pull requests welcomed for this? An option could be passed through to save contents to a variable.

**Comment by LarryKlugerDS** [comment]
Hi @cgsmith ,
Re: pull requests -- thank you for asking!

We use a customized version of the Swagger CodeGen app to create all of the SDKs. We're working on open sourcing our customizations.

**Today:** if you make a pull request then our engineering team can use your ideas to change the underlying models used to produce the SDK.

**In the future:** you'll be able to directly propose a PR that would change the model that would change the SDK.

Regards,
Larry

**Comment by LarryKlugerDS** [comment]
Good news: the engineering group reports that this issue has been fixed n the RC release 5.3.0-RC.

**If you can, please test the release candidate to check that the fix works for you. Thank you.**

The 5.3.0-RC (release candidate):
* source: https://github.com/docusign/docusign-php-client/tree/5.3.0-rc-v2.1-20.2.00 
* Packagist: https://packagist.org/packages/docusign/esign-client

The PRs:

https://github.com/docusign/docusign-php-client/pull/108
https://github.com/docusign/docusign-php-client/pull/109

**Comment by LarryKlugerDS** [comment]
No reply for 10 days. Closing. Please comment to re-open if there's still an issue. Thank you, Larry
